### PR TITLE
Implement forgot password form

### LINKED ISSUE #76: Implement "forgot password"
[8HR]
- [x] Require email
- [x] Set up mailservice

[https://github.com/Mentoraat/plannings-tool-server/issues/51](https://github.com/Mentoraat/plannings-tool-server/issues/51)

### PR DESCRIPTION
**Pull request by TimvdLippe**
TODO: Actually create the link and handle on the client.

Fixes #76

### CODE REVIEW

**Comment by TimvdLippe** [comment]
This feature is now ready. We have routing, ajax is loading correctly and all imports are fixed :tada:

File: app/elements/account-elements/account-page.html [hunk 1]
```diff
@@ -0,0 +1,97 @@
+<link rel="import" href="../../bower_components/paper-input/paper-input.html">
+<link rel="import" href="../../bower_components/paper-card/paper-card.html">
+<link rel="import" href="../../bower_components/paper-input/paper-input.html">
```
Comment (GijsWeterings) [inline app/elements/account-elements/account-page.html:2]: You're loading in paper-input twice (#L1)

File: app/elements/account-elements/forgot-password-request-page.html [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+          username: this.username.toLowerCase(),
+          email: this.email
+        };
+        this.$.account.generateRequest();
+      },
+      closeForgotPassword: function() {
+        this.selected = "";
+      },
+      handleForgotPasswordError: function(e) {
+        this.forgotPasswordError = false;
```
Comment (GijsWeterings) [inline app/elements/account-elements/forgot-password-request-page.html:46]: I'd rename this to `displayForgotPasswordError`, I read this as you resetting/overriding the error

File: app/elements/account-elements/login-page.html [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+        this.body = {
+          username: this.username.toLowerCase(),
+          password: this.password
+        };
+        this.$.account.generateRequest();
+      },
+
+      handleLoginError: function() {
+        this.validLogin = false;
```
Comment (GijsWeterings) [inline app/elements/account-elements/login-page.html:70]: and then for consistency, this should be renamed to something similar I guess

File: app/elements/account-elements/register-page.html [hunk 4]
```diff
@@ -0,0 +1,94 @@
+<link rel="import" href="../../bower_components/paper-input/paper-input.html">
+<link rel="import" href="../../bower_components/paper-button/paper-button.html">
+<link rel="import" href="account-page.html">
+
+<dom-module id="register-page">
+  <template>
+    <account-page id="account" sub-route="/register" heading="Register" body="[[body]]" on-iron-error="handleRegisterError">
```
Comment (GijsWeterings) [inline app/elements/account-elements/register-page.html]: Shouldn't register have some sort of succes event handler? Resetting to the login screen perhaps?

File: app/elements/account-elements/register-page.html [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+        this.fieldsFilled = true;
+
+        // If keyCode is [Enter], try register.
+        if (event.keyCode === 13) {
+          this.tryRegister();
+        }
+      },
+      tryRegister: function() {
+        if (this.password !== this.secondpassword) {
+          this.samePassword = false;
```
Comment (GijsWeterings) [inline app/elements/account-elements/register-page.html]: Now that we have it, we may as well reuse registerError and registerErrormessage, right?

File: app/elements/cardlist-api/card-list.html [hunk 6]
```diff
@@ -1,4 +1,8 @@
 <link rel="import" href="../../bower_components/paper-tabs/paper-tabs.html">
+<link rel="import" href="../../bower_components/paper-tabs/paper-tabs.html">
```
Comment (GijsWeterings) [inline app/elements/cardlist-api/card-list.html]: You're importing paper-tabs twice

File: bower.json [hunk 7]
```diff
@@ -12,15 +12,19 @@
...
   },
   "devDependencies": {
     "web-component-tester": "*",
     "test-fixture": "PolymerElements/test-fixture#^1.0.0"
   },
   "appPath": "app",
-  "moduleName": "planningtoolApp"
+  "moduleName": "planningtoolApp",
+  "resolutions": {
+    "paper-input": "spinner-mixin"
```
Comment (GijsWeterings) [inline bower.json:28]: wasn't this the other PR?

**Review by GijsWeterings — CHANGES_REQUESTED** [review]
few remarks, double imports etc.

File: bower.json (hunk 7 above)
Comment (TimvdLippe) [reply bower.json:28]: Yeah, but it was complaining on bower install. Can probably be removed after the other PR is merged and this PR is rebased.

**Review by GijsWeterings — APPROVED** [review]

**Pull request merged**